Tools/CISupport/build-webkit-org/steps.py

```diff
@@ -234,20 +234,27 @@ def __init__(self, repourl='https://github.com/WebKit/WebKit.git', **kwargs):
 
     def getResultSummary(self):
         revision = self.getProperty('got_revision')
-        self.setProperty('revision', revision[:HASH_LENGTH_TO_DISPLAY], 'CheckOutSource')
+        if revision is not None:
+            self.setProperty('revision', revision[:HASH_LENGTH_TO_DISPLAY], 'CheckOutSource')
 
-        if self.results == FAILURE:
-            self.build.addStepsAfterCurrentStep([CleanUpGitIndexLock()])
-
-        if self.results != SUCCESS:
-            return {'step': 'Failed to updated working directory'}
-        else:
+        if self.results == SUCCESS:
             return {'step': 'Cleaned and updated working directory'}
+        return {'step': 'Failed to updated working directory'}
 
     @defer.inlineCallbacks
     def run(self):
-        rc = yield super().run()
-        if rc == SUCCESS:
+        try:
+            # The "git fetch" command is executed by the git class with "abandonOnFailure=True"
+            # which means that if the command fails a BuildStepFailed exception is raised here.
+            rc = yield super().run()
+        except buildstep.BuildStepFailed:
+            rc = FAILURE
+        if rc == FAILURE:
+            if self.getProperty("cleanUpGitIndexLockAlreadyTried", False):
+                self.build.buildFinished(['Git issue'], FAILURE)
```

Comment:
It does the retry the build, but only once, retrying it always could lead to an infinite loop.

The login is the follow one:

1. Run step `clean-and-update-working-directory`
2. If step `clean-and-update-working-directory` fails then run step `clean-git-index-lock`
3. If step `clean-git-index-lock` fails then stop the build (as it failed to delete the `.git/index.lock` file and we don't know how to recover from that automatically)
4. If step `clean-git-index-lock` works then retry again step `clean-and-update-working-directory`.
5. If step `clean-and-update-working-directory` fails again then stop the build as we already tried to clean the `.git/index.lock` file but the step `clean-and-update-working-directory` keeps failing. That means there is an unknown issue with git that we don't know how to fix automatically and retrying again would likely cause an infinite loop.

Reply:
Thanks, that makes sense to me.

Our EWS implementation actually retries the entire build after deleting `.git/index.lock` - @JonWBedard do you know if there's a benefit to retrying the entire build instead of rerunning `clean-and-update-working-directory`? If not, then this PR LGTM.

Reply:
There isn't, @clopez is putting more thought into this that we did when in originally writing the step, I think.